File: rest-service/src/main/java/mb/lib/report/model/ReportPayload.java
package mb.lib.report.model;

import com.fasterxml.jackson.annotation.*;
import mb.api.model.io.JsonKeys;
import mb.lib.model.HashID;
import org.veupathdb.lib.blast.BlastFormatter;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class ReportPayload
{
  private HashID         jobID;
  private HashID         reportID;
  private BlastFormatter config;

  @JsonCreator
  public ReportPayload(
    @JsonProperty(JsonKeys.JobID)    HashID         jobID,
    @JsonProperty(JsonKeys.ReportID) HashID         reportID,
    @JsonProperty(JsonKeys.Config)   BlastFormatter config
  ) {
    this.jobID    = jobID;
    this.reportID = reportID;
    this.config   = config;
  }

  @JsonGetter(JsonKeys.JobID)
  public HashID getJobID() {
    return jobID;
  }

  @JsonSetter(JsonKeys.JobID)
  public ReportPayload setJobID(HashID jobID) {
    this.jobID = jobID;
    return this;
  }

  @JsonGetter(JsonKeys.Config)
  public BlastFormatter getConfig() {
    return config;
  }

  @JsonSetter(JsonKeys.Config)
  public ReportPayload setConfig(BlastFormatter config) {
    this.config = config;
    return this;
  }

  @JsonGetter(JsonKeys.ReportID)
  public HashID getReportID() {
    return reportID;
  }

  @JsonSetter(JsonKeys.ReportID)
  public ReportPayload setReportID(HashID reportID) {
    this.reportID = reportID;
    return this;
  }
}

File: rest-service/src/main/kotlin/mb/lib/report/model/ReportPayload.kt
package mb.lib.report.model

import com.fasterxml.jackson.annotation.JsonSetter
import com.fasterxml.jackson.annotation.JsonValue
import com.fasterxml.jackson.databind.node.ObjectNode
import mb.api.model.io.JsonKeys
import org.veupathdb.lib.blast.BlastFormatter
import org.veupathdb.lib.hash_id.HashID
import org.veupathdb.lib.jackson.Json
import org.veupathdb.lib.jackson.putIfNN

data class ReportPayload(
  @set:JsonSetter(JsonKeys.JobID)    var jobID:    HashID?         = null,
  @set:JsonSetter(JsonKeys.ReportID) var reportID: HashID?         = null,
  @set:JsonSetter(JsonKeys.Config)   var config:   BlastFormatter? = null,
) {

  @JsonValue
  fun toJson() = Json.new<ObjectNode> {
    putIfNN(JsonKeys.JobID, jobID) { it.string }
    putIfNN(JsonKeys.ReportID, reportID) { it.string }
    putIfNN(JsonKeys.Config, config) { it.toJSON().toJSON() }
  }
}